Runtime/ui/txt/paragraph.cs

```diff
@@ -1,1591 +1,1597 @@
-                    throw new ArgumentNullException("backingList");
-                }
-
-                this.tree = tree;
-                this.version = version;
-                this.backingList = backingList;
-            }
-
-            public int IndexOf(T item) {
-                if (this.tree.version != this.version) {
-                    throw new InvalidOperationException("The collection has been modified.");
-                }
-
-                return this.backingList.IndexOf(item);
-            }
-
-            public void Insert(int index, T item) {
-                throw new NotSupportedException();
-            }
-
-            public void RemoveAt(int index) {
-                throw new NotSupportedException();
-            }
-
-            public T this[int index] {
-                get {
-                    if (this.tree.version != this.version) {
-                        throw new InvalidOperationException("The collection has been modified.");
-                    }
-
-                    return this.backingList[index];
-                }
-                set { throw new NotSupportedException(); }
-            }
-
-            public void Add(T item) {
-                throw new NotSupportedException();
-            }
-
-            public void Clear() {
-                throw new NotSupportedException();
-            }
-
-            public bool Contains(T item) {
-                if (this.tree.version != this.version) {
-                    throw new InvalidOperationException("The collection has been modified.");
-                }
-
-                return this.backingList.Contains(item);
-            }
-
-            public void CopyTo(T[] array, int arrayIndex) {
-                if (this.tree.version != this.version) {
-                    throw new InvalidOperationException("The collection has been modified.");
-                }
-
-                this.backingList.CopyTo(array, arrayIndex);
-            }
-
-            public int Count {
-                get { return this.tree.count; }
-            }
-
-            public bool IsReadOnly {
-                get { return true; }
-            }
-
-            public bool Remove(T item) {
-                throw new NotSupportedException();
-            }
-
-            public IEnumerator<T> GetEnumerator() {
-                if (this.tree.version != this.version) {
-                    throw new InvalidOperationException("The collection has been modified.");
-                }
-
-                foreach (var item in this.backingList) {
-                    yield return item;
-                    if (this.tree.version != this.version) {
-                        throw new InvalidOperationException("The collection has been modified.");
-                    }
-                }
-            }
-
-            IEnumerator IEnumerable.GetEnumerator() {
-                return this.GetEnumerator();
-            }
-        }
-    }
+using System;
```

Comment:
huge changelist, make sure the changes are correct in this PR please

Reply:
The content looks almost the same, except the new method `AddAll()`. No idea why github marks them as different.